src/client/components/Sidebar.tsx

```diff
@@ -1,7 +1,117 @@
-import React from 'react'
+import { format, roundToNearestMinutes } from 'date-fns'
+import React, { useState } from 'react'
+import { useHistory } from 'react-router-dom'
 
+import MainButton from './Buttons'
+import { TimezonePicker } from './Inputs'
 import Sidetab from './Sidetab'
+import Spinner from './Spinner'
+import TimePicker from './Timepicker'
 
-export default function Sidebar(): JSX.Element {
-  return <Sidetab serverName="Eth History" />
+const TWENTYFOUR_HOURS_IN_MS = 1000 * 60 * 60 * 24
+export interface TimezoneObj {
+  value: string
+  label: string
+  abbrev: string
+  offset: number
+  altName: string
+}
+
+export const divider = {
+  marginTop: '15px',
+  width: '72%',
+  borderTop: '.5px solid white'
+}
+
+interface SidebarProps {
+  loading: boolean
+}
+
+export const initTimezone: TimezoneObj = {
+  value: 'GMT',
+  label: '(GMT+0:00) UTC',
+  abbrev: 'GMT',
+  offset: 0,
+  altName: 'GMT'
+}
+
+export function adjustDateToSelectedTimezone(
+  dateStr: string,
+  timezoneOffset: number
+): Date {
+  // Find offset between local timezone and UTC in milliseconds
+  const localTimezoneOffset = new Date().getTimezoneOffset() * 60 * 1000
+  // Find offset between selected timezone and UTC in milliseconds
+  const selectedTimezoneOffset = timezoneOffset * 60 * 60 * 1000
+  // Return adjusted date from local timezone to selected timezone
+  return new Date(
+    new Date(dateStr).getTime() - localTimezoneOffset - selectedTimezoneOffset
+  )
```

Comment:
See if you can change this from using JavaScript's Date object, to using date-fns instead. The `getTimezoneOffset()` method can return a number that is +/- 60 minutes because of Daylight Savings.